### PR TITLE
While statement

### PR DESCRIPTION
**Pull request by lushoBarlett**
Implements while loops, in accordance with the task graph.

As a side note, revisiting all those files made me realize that there's a heavy lack in formatting everywhere.

### CODE REVIEW

**Comment by SebastianMestre** [comment]
> As a side note, revisiting all those files made me realize that there's a heavy lack in formatting everywhere.

Yeah, I noticed that when touching `native.cpp` on my last pr. It might be cool to run clang-format project-wide and submit it as a pr

File: src/typed_ast.cpp [hunk 1]
```diff
@@ -171,6 +171,15 @@ TypedAST* convertAST(AST::ForStatement* ast) {
     return typed_for;
 }
 
+TypedAST* convertAST(AST::WhileStatement* ast) {
+    auto typed_for = new WhileStatement;
+
+    typed_for->m_condition = get_unique(ast->m_condition);
+    typed_for->m_body      = get_unique(ast->m_body);
+
+    return typed_for;
```
Comment (SebastianMestre) [inline src/typed_ast.cpp:180]: it's a bit ugly that the variable is named `typed_for`. either `typed_while` or `typed_ast` would be more appropriate

Comment (lushoBarlett) [reply src/typed_ast.cpp:180]: Woops, missed that one

**Comment by lushoBarlett** [comment]
> > As a side note, revisiting all those files made me realize that there's a heavy lack in formatting everywhere.
> 
> Yeah, I noticed that when touching `native.cpp` on my last pr. It might be cool to run clang-format project-wide and submit it as a pr

I was gonna suggest that exact thing, so if you would like to do it, since you already have it installed, you may proceed.

**Pull request merged**